modules/nlu/src/backend/election/simple-election.ts

```diff
@@ -0,0 +1,99 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+
+const NONE_INTENT = 'none'
+
+function detectAmbiguity(intents: sdk.NLU.Intent[]): boolean {
+  // +- 10% away from perfect median leads to ambiguity
+  const preds = intents
+  const perfectConfusion = 1 / preds.length
+  const low = perfectConfusion - 0.1
+  const up = perfectConfusion + 0.1
+  const confidenceVec = preds.map(p => p.confidence)
+
+  const allInRange = (vec: number[], lower: number, upper: number) =>
+    vec.map(v => _.inRange(v, lower, upper)).every(_.identity)
+
+  const ambiguous =
+    preds.length > 1 &&
+    (allInRange(confidenceVec, low, up) ||
+      (preds[0].name === NONE_INTENT && allInRange(confidenceVec.slice(1), low, up)))
+
+  return ambiguous
+}
+
+/**
+ * @description confidence of none intent becomes Math.max(none, oos)
+ * @param oos confidence of OOS
+ */
+const replaceNone = (oos: number) => (i: sdk.NLU.Intent): sdk.NLU.Intent => {
+  if (i.name !== NONE_INTENT) {
+    return i
+  }
+  return { ...i, confidence: Math.max(i.confidence, oos) }
+}
+
+const toNLUIntent = (context: string) => (intent: sdk.NLU.ContextPrediction['intents'][number]): sdk.NLU.Intent => {
+  return {
+    name: intent.label,
+    confidence: intent.confidence,
+    context
+  }
+}
+
+/**
+ * @description make probabilities sum up to 1 (because, oos classifier is different from intent classifier, probabilities don't sum up to 1)
+ * @param oos confidence of OOS
+ */
+const makeTotalConfidenceOne = (intents: sdk.NLU.Intent[]): sdk.NLU.Intent[] => {
```

Comment:
This step should actually be done inside nlu-core, but by doing so, we would the legacy-election behavior...